mediator/tenants/models.py

```diff
@@ -0,0 +1,38 @@
+from cryptography.fernet import Fernet
+from django.conf import settings
+from django.contrib.auth.models import AbstractUser
+from django.db import models
+
+
+class Tenant(models.Model):
+    short_name = models.SlugField(max_length=200, unique=True)
```

Comment:
There are pros and cons to this approach, but could consider making this the primary key. Then it's a bit easier in the other models to know what `Tenant` something belongs to without having to do any joins. Think main disadvantage is just that keys and indices take up more storage space.

Reply:
I know that's the approach we've taken in HQ, and I think when data spans multiple databases, that really is the best approach. But it does make renaming the key a headache, and probably more so from Django Admin. I know we simply forbid it in HQ, but we end up with empty project spaces. Using [select_related](https://docs.djangoproject.com/en/2.1/ref/models/querysets/#select-related) can do the join automatically, mitigate the overhead of fetching the Tenant record separately, and the value would always be available.

A compromise might be to keep an automatic primary key, and use `short_name` as the foreign key field on `Upstream` ... but renames would still need to be done in multiple places, and still probably more complicated than necessary in Django Admin.

(Can I use the second paragraph of this [SO answer](https://stackoverflow.com/a/2056129) to support my opinion?)

Reply:
Hmm, yeah that's true and that SO post is a compelling argument. I'm convinced!